src/server/client.ts

```diff
@@ -0,0 +1,505 @@
+/// <reference path="session.ts" />
+
+module ts.server {
+
+    export interface SessionClientHost extends LanguageServiceHost {
+        writeMessage(message: string): void;
+    }
+
+    interface CompletionEntry extends CompletionInfo {
+        fileName: string;
+        position: number;
+    }
+
+    interface RenameEntry extends RenameInfo {

```

Comment:
This worries me.  Getting the locations up front means this is far costlier.  That can lead users down a pit of fail.